src/main/java/Duke.java

```diff
@@ -21,14 +22,30 @@ private static void echo() {
         // get user input
         line = in.nextLine();
         while (!line.equals("bye")) {
+            if (line.equals("list")) {
+                showList();
+                line = in.nextLine();
+                continue;
+            }
             System.out.println("    ____________________________________________________________");
-            System.out.println("     " + line);
+            System.out.println("added: " + line);
+            descriptions.add(line);
             System.out.println("    ____________________________________________________________");
             line = in.nextLine();
         }
+        if (line.equals("bye")) {
+            System.out.println("    ____________________________________________________________");
+            System.out.println("     Bye. Hope to see you again soon!");
+            System.out.println("    ____________________________________________________________");
+        }
+    }
+
+    private static void showList() {
         System.out.println("    ____________________________________________________________");
-        System.out.println("     Bye. Hope to see you again soon!");
-        System.out.println("    ____________________________________________________________");
+        System.out.println("     Here are the tasks in your list:");
+        for (int i = 0; i < descriptions.size(); i++) {
+            System.out.println(i+1 + "." + descriptions.get(i));
```

Comment:
There should always be spaces before and after +

Reply:
Thank you for pointing it out, I will work on it!